app/queries/action_page_filters.rb

```diff
@@ -0,0 +1,62 @@
+class ActionPageFilters
+  def self.run(**data)
+    new(**data).run
+  end
+
+  def initialize(relation: ActionPage.all, **filters)
+    @relation = relation
+    @filters = filters
+  end
+
+  def run
+    process_date_range
+    filters.delete :date_range
+
+    filters.each do |f, val|
+      next unless valid_query?(f, val)
+      @relation = if NAMED_SCOPES.include? f
+                    relation.send(f, val)
+                  else
+                    relation.where(f => val)
+                  end
+    end
+    relation
+  end
+
+  private
+
+  NAMED_SCOPES = %i(type status).freeze
+  VALID_FILTERS = %i(type status author category).freeze
+
+  attr_accessor :relation, :filters
+
+  def default_date_range?(start_date, end_date)
+    start_date == end_date && start_date == Time.zone.today
```

Comment:
I'm not happy with how this is handled -- basically this checks to see if the date range given is the default, and doesn't run the date query if it is. 

Ideally the date picker would be blank by default and it would be easier for the user to clear it out entirely, but I wasn't sure how to do that.

Reply:
It seems to work to wait on initializing the widget until the input is focused, like this:

```js
  $('#filter_action_pages #date_range').on('focus', function() {
    $(this).daterangepicker({
      locale: { format: 'YYYY-MM-DD' }
    });
  });
```

Then listen for the `apply.daterangepicker` event instead of `change` to avoid some weird resulting behavior.

In general though if you have to detect/remove default parameters it's nice to do that in a before_action.